cylc/flow/scripts/workflow_state.py

```diff
@@ -18,35 +18,55 @@
 
 r"""cylc workflow-state [OPTIONS] ARGS
 
-Retrieve task states from the workflow database.
+Print or poll for task states or outputs in a workflow database.
 
-Print task states retrieved from a workflow database; or (with --task,
---point, and --status) poll until a given task reaches a given state; or (with
---task, --point, and --message) poll until a task receives a given message.
-Polling is configurable with --interval and --max-polls; for a one-off
-check use --max-polls=1. The workflow database does not need to exist at
-the time polling commences but allocated polls are consumed waiting for
-it (consider max-polls*interval as an overall timeout).
+For specific cycle/task instances poll until the given status or output is
+achieved (command success) or the max number of polls is reached (failure).
 
-Note for non-cycling tasks --point=1 must be provided.
+For less specific queries, immediate results are printed (no polling is done).
 
-For your own workflows the database location is determined by your
-site/user config. For other workflows, e.g. those owned by others, or
-mirrored workflow databases, use --run-dir=DIR to specify the location.
+Override default polling parameters with --max-polls and --interval.
+
+If the database does not exist at first, polls are consumed waiting for it.
+
+For task outputs, give the task message, not the output label.
+
+For non-cycling workflows, provide --point=1 for specific queries.
+
+This command can be used to make polling tasks that trigger off of tasks in
+other workflows - but see also the built-in workflow_state xtrigger for that.
```

Comment:
I'm more acknowledging the fact that some users already do this, here at least (although it may be a pre-xtrigger hang-over). Maybe there are valid use cases - a task that uses the command could do other things at the same time. 

Your wording isn't quite correct as the task is not a "polling task" anymore if the xtrigger is doing the polling, but I'll change the wording to suggest that this is possible but not necessarily desirable.

Reply:
> Not sure it is a good idea to suggest using this command for this purpose at all? Is there any circumstance the xtrigger is not better for it?

Yes, they don't clog up the workflow's process pool, can be more efficient and can be used to poll workflows on remote platforms which xtriggers cannot do.